lib/pure/options.nim

```diff
@@ -338,15 +310,17 @@ proc `==`*(a, b: Option): bool {.inline.} =
     assert b == d
     assert not (a == b)
 
-  (a.isSome and b.isSome and a.val == b.val) or (not a.isSome and not b.isSome)
+  when T is SomePointer:
+    a.val == b.val
+  else:
+    (a.isSome and b.isSome and a.val == b.val) or (a.isNone and b.isNone)
```

Comment:
This isn't just a doc change, maybe it should go into a separate PR?

Reply:
Hmm, that was suggested by @timotheecour, however I don't think such a small change warrants a new PR. What if I just change the PR description?

Reply:
i had suggested that change; I think it's simple enough to be grouped in same PR but I don't mind either way